### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by jvisker**
Refactoring to remove component modules. We will need to talk about this in the Terraform working group for some of the the changes.

### CODE REVIEW

**Comment by jvisker** [comment]
I converted it to a draft. This PR right now is going to be used for a conversation in today's meeting. I'm not a hundred percent on the contract so I haven't updated the readme yet.

**Comment by jvisker** [comment]
I have now pulled out the hosted zone and have tested

**Comment by jvisker** [comment]
I'll work on the readme now though

**Comment by jvisker** [comment]
The readme should at least be passable now.

File: examples/example.tf [hunk 1]
```diff
@@ -3,10 +3,25 @@ provider "aws" {
   region  = "us-west-2"
 }
 
+provider "aws" {
+  alias  = "aws_n_va"
+  region = "us-east-1"
+}
+
+data "aws_route53_zone" "zone" {
+  provider = aws.aws_n_va
```
Comment (yoshutch) [inline examples/example.tf:12]: maybe not needed?

Comment (yoshutch) [reply examples/example.tf]: @jvisker done I think. Want to double check?

Comment (jvisker) [reply examples/example.tf]: Looks good

**Pull request merged**